x/emissions/keeper/keeper.go

```diff
@@ -2070,6 +2081,37 @@ func (k *Keeper) IsReputerRegisteredInTopic(ctx context.Context, topicId TopicId
 	return k.topicReputers.Has(ctx, topicKey)
 }
 
+// wrapper for set operation around activeTopics
+func (k *Keeper) SetActiveTopics(ctx context.Context, topicId TopicId) error {
+	if err := types.ValidateTopicId(topicId); err != nil {
+		return err
+	}
+	return k.activeTopics.Set(ctx, topicId)
+}
+
+// wrapper for set operation around blockToActiveTopics
+func (k *Keeper) SetBlockToActiveTopics(ctx context.Context, block BlockHeight, topicIds types.TopicIds) error {
+	if err := types.ValidateBlockHeight(block); err != nil {
+		return err
+	}
+	for _, topicId := range topicIds.TopicIds {
+		if err := types.ValidateTopicId(topicId); err != nil {
+			return err
+		}
+	}
+	return k.blockToActiveTopics.Set(ctx, block, topicIds)
+}
+
+func (k *Keeper) SetTopicToNextPossibleChurningBlock(ctx context.Context, topicId TopicId, block BlockHeight) error {
+	if err := types.ValidateTopicId(topicId); err != nil {
+		return err
+	}
```

Comment:
How can we ever get a topic id == 0 in these situations?

Reply:
This paranoia can go far xD But yeah, if we are planning to optimize this later, it's fine by me. It can be a pain in terms of maintenance if we keep this pattern all around. Also, I thought we'd just focus on values that are used in calculations and are suffering mutations... but again, if we plan to change this in the future, that's ok for now.

Reply:
yeah I think we will be able to remove some validations from other places if we want to keep them at the keeper

but yes I recognize not all of these validations are strictly necessary in all cases

but being thorough is better, after all we are building a financial system